lib/server-starter.js

```diff
@@ -86,7 +90,23 @@ class Server extends EventEmitter {
       this.emit('exit', code, signal);
     });
 
-    return new Promise(resolve => this._srv.close(resolve));
+    return new Promise(resolve => this._srv.close(
+      () => {
+        if (this._useFdPass) resolve();
+        else {
+          const now = new Date();
+          const timeToStop = new Date(now.getTime() + connectTimeout);
```

Comment:
Even my feedback on the unused `now` constant from the previous PR you've just dismissed without changes.